Framework/PythonInterface/plugins/algorithms/VesuvioTOFFit.py

```diff
@@ -112,17 +111,22 @@ def _run_fit_impl(self, data_ws, workspace_index, fit_options, simulation=False)
             function_str = fit_options.create_function_str()
             constraints = fit_options.create_constraints_str()
             ties = fit_options.create_ties_str()
-            results = self._do_fit(function_str, data_ws, workspace_index, constraints, ties,
-                                   max_iter=self.getProperty("MaxIterations").value)
+
+            chi_square, params, fitted_data = self._do_fit(function_str, data_ws, workspace_index,
+                                                       constraints, ties,
+                                                       max_iter=self.getProperty("MaxIterations").value)
 
             # Run second time using standard CompositeFunction & no constraints matrix to
             # calculate correct reduced chisq
-            param_values = TableWorkspaceDictionaryFacade(results[1])
+            param_values = TableWorkspaceDictionaryFacade(params)
 
         function_str = fit_options.create_function_str(param_values)
         max_iter = 0 if simulation else 1
-        return self._do_fit(function_str, data_ws, workspace_index, constraints,
-                            ties, max_iter=max_iter)
+        reduced_chi_square, _, _= self._do_fit(function_str, data_ws,
+                                               workspace_index, constraints,
+                                               ties, max_iter=max_iter)
+
+        return chi_square, params, fitted_data
```

Comment:
Did you mean to return the `reduced_chi_square` here and not `chi_square`?

It looks like `reduced_chi_square` is not used otherwise.